include/gunrock/algorithms/tc.hxx

```diff
@@ -0,0 +1,152 @@
+    auto P = this->get_problem();
+    auto n_vertices = P->get_graph().get_number_of_vertices();
+
+    f->sequence((vertex_t)0, n_vertices, context.get_context(0)->stream());
+  }
+
+  void loop(gcuda::multi_context_t& context) override {
+    // Data slice
+    auto E = this->get_enactor();
+    auto P = this->get_problem();
+    auto G = P->get_graph();
+
+    auto vertex_triangles_count = P->result.vertex_triangles_count;
+    auto iteration = this->iteration;
+
+    auto intersect = [G, vertex_triangles_count] __host__ __device__(
+                         vertex_t const& source,    // ... source
+                         vertex_t const& neighbor,  // neighbor
+                         edge_t const& edge,        // edge
+                         weight_t const& weight     // weight (tuple).
+                         ) -> bool {
+      if (neighbor > source) {
+        auto src_vertex_triangles_count = G.get_intersection_count(
+            source, neighbor,
+            [vertex_triangles_count](auto intersection_vertex) {
+              math::atomic::add(&(vertex_triangles_count[intersection_vertex]),
+                                vertex_t{1});
+            });
+      }
+      return false;
+    };
+
+    // Execute advance operator on the provided lambda
+    operators::advance::execute<operators::load_balance_t::block_mapped>(
+        G, E, intersect, context);
+  }
+};  // struct enactor_t
+
+template <typename graph_t>
+float run(graph_t& G,
+          bool reduce_all_triangles,
+          typename graph_t::vertex_type* vertex_triangles_count,  // Output
+          std::size_t* total_triangles_count,                     // Output
+          std::shared_ptr<gcuda::multi_context_t> context =
+              std::shared_ptr<gcuda::multi_context_t>(
+                  new gcuda::multi_context_t(0))  // Context
+) {
+  // <user-defined>
+  using vertex_t = typename graph_t::vertex_type;
+  using weight_t = typename graph_t::weight_type;
+
+  using param_type = param_t<vertex_t>;
+  using result_type = result_t<vertex_t>;
+
+  param_type param(reduce_all_triangles);
+  result_type result(vertex_triangles_count, total_triangles_count);
+  // </user-defined>
+
+  using problem_type = problem_t<graph_t, param_type, result_type>;
+  using enactor_type = enactor_t<problem_type>;
+
+  problem_type problem(G, param, result, context);
+  problem.init();
+  problem.reset();
+
+  enactor_type enactor(&problem, context);
+  auto time = enactor.enact();
+
+  if (param.reduce_all_triangles) {
+    auto policy = context->get_context(0)->execution_policy();
+    *result.total_triangles_count = thrust::transform_reduce(
+        policy, result.vertex_triangles_count,
+        result.vertex_triangles_count + G.get_number_of_vertices(),
+        [] __device__(const vertex_t& vertex_triangles) {
+          return static_cast<std::size_t>(vertex_triangles);
+        },
+        std::size_t{0}, thrust::plus<std::size_t>());
+  }
```

Comment:
Should be timed/be a part of the enactor's `is_converged`? (see: https://github.com/gunrock/essentials/blob/dev/include/gunrock/algorithms/pr.hxx#L155)

Reply:
Resolved https://github.com/gunrock/essentials/pull/140/commits/2b0874249a2d57b821842f3bbf251a47abb72d39